### PR TITLE
DCD-889: Add version tagging and ASG tag permissions

### PR DESCRIPTION
**Pull request by tarka**
This adds commit & timestamp tagging to the Jira QS, and grants the nodes the ability to manipulate the ASG tags to persist run-time metadata.

### CODE REVIEW

**Review by theghostwhoforks — APPROVED** [review]
Is it necessarry to create the tag at each level? Is there an equivalent of us creating this at the top stack level and CFn somehow propagates it down?

**Comment by tarka** [comment]
Not that I can see. Tagging is inconsistent in AWS; not all resources support tagging, and the format is different in some places. Only the ASG supports propagating tags.

File: templates/quickstart-jira-dc.template.yaml [hunk 1]
```diff
@@ -1002,6 +1005,13 @@ Resources:
         - Key: Cluster
           Value: !Ref AWS::StackName
           PropagateAtLaunch: true
+        # NOTE: The leading COMMIT/TIMESTAMP are used to locate the position to update; see scripts/update-tags.py
```
Comment (bpar476) [inline templates/quickstart-jira-dc.template.yaml:1009]: We don't have the script to update the commit/timestamp in this repo. Should we update the comment to say use the script from Atlassian services or should we duplicate the script?

**Comment by tarka** [comment]
Once the services PR is on master we can update the submodule and call it from any of the quickstarts. I'll also add a pre-commit hook.

**Review by bpar476 — APPROVED** [review]
Sounds good

**Pull request merged**